pykern/pkcli/test.py

```diff
@@ -171,31 +173,57 @@ def _except(exc, output, restartable):
                     return "restart"
             return _fail(output)
 
-        def _skipped(ouput):
-            return _TEST_SKIPPED.findall(ouput)
-
         def _fail(output):
             self.failures.append(output)
             return f"FAIL {output}"
 
+        def _ignore_warnings():
+            if not _cfg.ignore_warnings:
+                return []
+            return list(
+                # This undoes what pytest does
+                itertools.chain.from_iterable(
```

Comment:
Would list.extend() be better than chain.from_iterable()?

Also, I remember running into problems trying to do ("-W", f"error::{runtime_warning_and_regex}"), but looking at how you pulled this off I think it might be worth another try.

Reply:
chain.from_iterable flattens the list by one level. sum reduces a list. list.extend would require a temporary. It's debatable. I like functional that eliminates a temporary.

Reply:
It's better 272f257e85dbb3accef7f59de69d94fc1f7bc328.